lib/ansible/module_utils/aws/elbv2.py

```diff
@@ -0,0 +1,737 @@
+
+        # Required parameters
+        params = dict()
+        params['Name'] = self.name
+        params['Type'] = self.type
+
+        # Other parameters
+        if self.subnets is not None:
+            params['Subnets'] = self.subnets
+        if self.security_groups is not None:
+            params['SecurityGroups'] = self.security_groups
+        params['Scheme'] = self.scheme
+        if self.tags is not None:
+            params['Tags'] = self.tags
+
+        try:
+            self.elb = self.connection.create_load_balancer(**params)['LoadBalancers'][0]
+            self.changed = True
+            self.new_load_balancer = True
+        except (BotoCoreError, ClientError) as e:
+            self.module.fail_json_aws(e, msg=str(e))
+
+        if self.wait:
+            status_achieved, new_elb = self.wait_for_status('active', self.name)
+
+    def modify_elb_attributes(self):
+        """
+        Update ELB attributes if required
+        :return:
+        """
+
+        update_attributes = []
+
+        if self.access_logs_enabled and self.elb_attributes['access_logs_s3_enabled'] != "true":
+            update_attributes.append({'Key': 'access_logs.s3.enabled', 'Value': "true"})
+        if not self.access_logs_enabled and self.elb_attributes['access_logs_s3_enabled'] != "false":
+            update_attributes.append({'Key': 'access_logs.s3.enabled', 'Value': 'false'})
+        if self.access_logs_s3_bucket is not None and self.access_logs_s3_bucket != self.elb_attributes['access_logs_s3_bucket']:
+            update_attributes.append({'Key': 'access_logs.s3.bucket', 'Value': self.access_logs_s3_bucket})
+        if self.access_logs_s3_prefix is not None and self.access_logs_s3_prefix != self.elb_attributes['access_logs_s3_prefix']:
+            update_attributes.append({'Key': 'access_logs.s3.prefix', 'Value': self.access_logs_s3_prefix})
+        if self.deletion_protection and self.elb_attributes['deletion_protection_enabled'] != "true":
+            update_attributes.append({'Key': 'deletion_protection.enabled', 'Value': "true"})
+        if self.deletion_protection is not None and not self.deletion_protection and self.elb_attributes['deletion_protection_enabled'] != "false":
+            update_attributes.append({'Key': 'deletion_protection.enabled', 'Value': "false"})
+        if self.idle_timeout is not None and str(self.idle_timeout) != self.elb_attributes['idle_timeout_timeout_seconds']:
+            update_attributes.append({'Key': 'idle_timeout.timeout_seconds', 'Value': str(self.idle_timeout)})
+
+        if update_attributes:
+            try:
+                self.connection.modify_load_balancer_attributes(LoadBalancerArn=self.elb['LoadBalancerArn'], Attributes=update_attributes)
+                self.changed = True
+            except (BotoCoreError, ClientError) as e:
+                # Something went wrong setting attributes. If this ELB was created during this task, delete it to leave a consistent state
```

Comment:
I chose this behaviour because if the ELB is left there then the user will start paying for it and it may not be expected considering they saw the task fail.

When you say you think the failure message should indicate what happened do you mean wrap the delete_load_balancer call in another `try`?  Or just a generic message saying 'We created the load balancer but failed to set attributes so we removed it'?